test/suites/openfga.sh

```diff
@@ -0,0 +1,282 @@
+  [ "$(lxc_remote network acl list candid-openfga: -f csv | wc -l)" = 0 ]
+  ! lxc_remote network acl create candid-openfga:test-acl || false
+  lxc network acl rm acl1
+
+  # Should not be able to see or create network zones.
+  lxc network zone create zone1
+  [ "$(lxc_remote network zone list candid-openfga: -f csv | wc -l)" = 0 ]
+  ! lxc_remote network zone create candid-openfga:test-zone || false
+  lxc network zone rm zone1
+
+  # Should not be able to see or create profiles.
+  [ "$(lxc_remote profile list candid-openfga: -f csv | wc -l)" = 0 ]
+  ! lxc_remote profile create candid-openfga:test-profile || false
+
+  # Should not be able to see or create image aliases
+  test_image_fingerprint="$(lxc image info testimage | awk '/^Fingerprint/ {print $2}')"
+  [ "$(lxc_remote image alias list candid-openfga: -f csv | wc -l)" = 0 ]
+  ! lxc_remote image alias create candid-openfga:testimage2 "${test_image_fingerprint}" || false
+
+  # Should not be able to see or create storage pool volumes.
+  pool_name="$(lxc_remote storage list candid-openfga: -f csv | cut -d, -f1)"
+  lxc storage volume create "${pool_name}" vol1
+  [ "$(lxc_remote storage volume list "candid-openfga:${pool_name}" -f csv | wc -l)" = 0 ]
+  [ "$(lxc_remote storage volume list "candid-openfga:${pool_name}" --all-projects -f csv | wc -l)" = 0 ]
+  ! lxc_remote storage volume create "candid-openfga:${pool_name}" test-volume || false
+  lxc storage volume rm "${pool_name}" vol1
+
+  # Should not be able to see any operations.
+  [ "$(lxc_remote operation list candid-openfga: -f csv | wc -l)" = 0 ]
+  [ "$(lxc_remote operation list candid-openfga: --all-projects -f csv | wc -l)" = 0 ]
+
+  # Image list will still work but none will be shown because none are public.
+  [ "$(lxc_remote image list candid-openfga: -f csv | wc -l)" = 0 ]
+
+  # Image edit will fail. Note that this fails with "not found" because we fail to resolve the alias (image is not public
+  # so it is not returned from the DB).
+  ! lxc_remote image set-property candid-openfga:testimage requirements.secureboot true || false
+  test_image_fingerprint_short="$(echo "${test_image_fingerprint}" | cut -c1-12)"
+  ! lxc_remote image set-property "candid-openfga:${test_image_fingerprint_short}" requirements.secureboot true || false
+
+  # Should be able to list public images.
+  lxc image show testimage | sed -e "s/public: false/public: true/" | lxc image edit testimage
+  lxc_remote image list candid-openfga: -f csv | grep -Fq "${test_image_fingerprint_short}"
+  lxc image show testimage | sed -e "s/public: true/public: false/" | lxc image edit testimage
+}
+
+user_is_instance_user() {
+  # shellcheck disable=SC3043
+  local instance_name
+  instance_name="${1}"
+
+  # Check we can still interact with the instance.
+  LXC_LOCAL='' lxc_remote exec "candid-openfga:${instance_name}" -- echo "bar"
+  touch "${TEST_DIR}/tmp"
+  lxc_remote file push "${TEST_DIR}/tmp" "candid-openfga:${instance_name}/root/tmpfile.txt"
```

Comment:
I think it'd be safer to remove that `/root/tmpfile.txt`. If you do the removal with a `lxc exec ... -- rm /root/tmpfile.txt` you can drop the earlier `exec ... -- echo "bar"`  test.